packages/database/src/migrations/20230203051720-ConvertReportsToPullFromDataTables-modifies-schema.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+var dbm;
+var type;
+var seed;
+
+/**
+ * We receive the dbmigrate dependency from dbmigrate initially.
+ * This enables us to not have to rely on NODE_PATH.
+ */
+exports.setup = function (options, seedLink) {
+  dbm = options.dbmigrate;
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+const convertFetchDataTransformToUseDataTables = transform => {
+  const { parameters } = transform;
+  // If passing both data group as param, then fetching events
+  if (parameters.dataGroup) {
+    return { ...transform, dataTableCode: 'events' };
+  }
+
+  return { ...transform, dataTableCode: 'analytics' };
+};
+
+const revertFetchDataTransformFromUsingDataTables = transform => {
+  const { dataTableCode, ...restOfTransform } = transform;
+  return { ...restOfTransform };
+};
+
+const convertToNewConfig = config => {
+  const { transform, ...restOfConfig } = config;
+
+  if (!transform) {
+    return config;
+  }
+
+  const newTransform = transform.map(tform => {
+    if (tform.transform === 'fetchData') {
+      return convertFetchDataTransformToUseDataTables(tform);
```

Comment:
I don't think `.map` runs in parallel, right? `.map` will always run each item in order. It can get a bit funky if you're using `async` code and the side effects of the function occur in a hard to predict order. But everything here is synchronous code

Reply:
My bad, that confused me before but not anymore. Thanks!